src/labone/core/shf_vector_data.py

```diff
@@ -484,7 +492,12 @@ def _deserialize_shf_scope_vector(
     """
     # Parse header
     raw_extra_header = raw_data[:header_length]
-    version = _parse_extra_header_version(extra_header_info)
+    try:
+        version = _parse_extra_header_version(extra_header_info)
+    except ValueError:
```

Comment:
`version` is undefined in the later code if `raw_data` has `len > 0`.
If the code ends here, is `raw_data` always length 0, or what should happen if it is not.

Reply:
The catch block returns. Or what am I missing

Reply:
if `ValueError` is caught and `len(raw_data)` is not `0`, it will continue to:

```
   extra_header = ShfScopeVectorExtraHeader.from_binary(
         raw_extra_header,
         version=version,
```

and `version` is not then specified.

Reply:
You are absolutely right ... I missed that. I pushed a change that raised an error in this particular case.